### PR TITLE
Feature bank card registration

### PR DESCRIPTION
**Pull request by jwaguet**

### CODE REVIEW

**Comment by jwaguet** [comment]
@sylfabre PR stp

File: src/Manager/BankCardManager.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Creating bank card
+     * @param $bankCardRegistration
+     * @return BankCardRegistration
+     */
+    public function createBankCardRegistration($bankCardRegistration): BankCardRegistration
```
Comment (sylfabre) [inline src/Manager/BankCardManager.php]: missing argument type

File: src/Object/BankCardRegistration.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+/**
+ * Online payment with a card on the S-Money hosted form.
+ * This object can also describe a refund.
+ */
+class BankCardRegistration extends AbstractHydratable
+{
+    /**
+     * Bank card information for the registration
+     *
+     * @var int
```
Comment (sylfabre) [inline src/Object/BankCardRegistration.php]: wrong type

File: src/Object/BankCardRegistration.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+     * Error code for the registration
+     *
+     * @var string
+     */
+    public $errorCode;
+
+    /**
+     * Return codes detail's
+     *
+     * @var string
```
Comment (sylfabre) [inline src/Object/BankCardRegistration.php]: isn't it array?

File: src/Object/BankCardRegistration.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    public const CARDTYPE_CB = 'CB';
+    public const CARDTYPE_MASTERCARD = 'MASTERCARD';
+    public const CARDTYPE_MAESTRO = 'MAESTRO';
+    public const CARDTYPE_VISA = 'VISA';
+    public const CARDTYPE_ELECTRON = 'VISA_ELECTRON';
+
+    /**
+     * List of available cards
+     *
+     * @var int
```
Comment (sylfabre) [inline src/Object/BankCardRegistration.php]: wrong type

File: src/Manager/BankCardManager.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        $path = '/cards/registrations';
+        $method = RequestMethodInterface::METHOD_POST;
+
+        $data = [
+            'card' => [
+                'appcardid' => $bankCardRegistration->card['appCardId'],
```
Comment (sylfabre) [inline src/Manager/BankCardManager.php]: use a DTO for `$bankCardRegistration->card` instead of an array

**Pull request merged**